### PR TITLE
Fixed bug where setting waveform acquisition to get all points would …

### PR DESCRIPTION
**Pull request by fthouin**
…return none at all

### CODE REVIEW

File: pymeasure/instruments/siglenttechnologies/siglent_sds1072cml.py [hunk 1]
```diff
@@ -65,13 +65,23 @@ def get_waveform(self):
...
-                response,
-                offset=descriptor_dictionary["descriptorOffset"],
-            ),
-        )
+        if descriptor_dictionary["numDataPoints"] == 0:
+            rawWaveform = list(
+                struct.unpack_from(
+                    "%db" % descriptor_dictionary["pointsScreen"],
+                    response,
+                    offset=descriptor_dictionary["descriptorOffset"],
```
Comment (BenediktBurger) [inline pymeasure/instruments/siglenttechnologies/siglent_sds1072cml.py:73]: If all data points are returned, should not the offset be 0?

File: pymeasure/instruments/siglenttechnologies/siglent_sds1072cml.py [hunk 2]
```diff
@@ -65,13 +65,23 @@ def get_waveform(self):
...
+                ),
+            )
+        else:
+            rawWaveform = list(
+                struct.unpack_from(
+                    "%db" % descriptor_dictionary["numDataPoints"],
```
Comment (BenediktBurger) [inline pymeasure/instruments/siglenttechnologies/siglent_sds1072cml.py:79]: another idea to not repeat most of the code:
~~~python
data_points = descriptor_dictionary["numDataPoints"] if descriptor_dictionary["numDataPoints"] els escriptor_dictionary["pointsScreen"]
raw_waveform = list(... f"{data_points:d}b", response,...)
~~~

**Review by BenediktBurger — COMMENTED** [review]
Thanks for your bug fix.

A suggestion for smaller code and a question regarding working code.

File: pymeasure/instruments/siglenttechnologies/siglent_sds1072cml.py (hunk 1 above)
Comment (fthouin) [reply pymeasure/instruments/siglenttechnologies/siglent_sds1072cml.py:73]: The first bytes are a descriptor string and do not contain data. This is why they are skipped.

File: pymeasure/instruments/siglenttechnologies/siglent_sds1072cml.py (hunk 2 above)
Comment (fthouin) [reply pymeasure/instruments/siglenttechnologies/siglent_sds1072cml.py]: Thanks for the suggestion. I implemented it in the code.

**Comment by BenediktBurger** [comment]
Thanks for your work.

**Pull request merged**